packages/pigeon/lib/ast.dart

```diff
@@ -105,6 +108,24 @@ class TypeDeclaration {
   /// True if the type is nullable.
   final bool isNullable;
 
+  @override
+  int get hashCode {
+    // This has to be implemented because TypeDeclaration is used as a Key to a
+    // Map in generator_tools.dart.
+    int hash = 17;
+    hash = hash * 37 + baseName.hashCode;
+    hash = hash * 37 + isNullable.hashCode;
+    for (final TypeDeclaration typeArgument in typeArguments) {
+      hash = hash * 37 + typeArgument.hashCode;
+    }
+    return hash;
```

Comment:
Isn't `Object.hash` the canonical way of doing this as of 2.14?

Reply:
Looks like `Object.hashAll` from dart would fit? (or its analogue `hashList` in flutter)

Something like this

```dart
return hashValues(
  baseName,
  isNullable,
  hashList(typeArguments),
);
```

Reply:
That requires a newer version of Dart.